src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -28,18 +29,33 @@ public CollectionTallyResult(List<object> missingItems, List<object> extraItems)
 
         private readonly NUnitEqualityComparer comparer;
 
+        private bool _sorted = false;
+
         /// <summary>The result of the comparison between the two collections.</summary>
         public CollectionTallyResult Result
         {
             get
             {
-                return new CollectionTallyResult(
-                    new List<object>(_missingItems),
-                    new List<object>(_extraItems));
+                var missingItems = new List<object>(_missingItems.Count);
+                foreach (var o in _missingItems)
+                    missingItems.Add(o);
+
+                List<object> extraItems = new List<object>();
+                if(_sorted)
+                {
+                    for (int index = _extraItems.Count - 1; index >= 0; index--)
+                        extraItems.Add(_extraItems[index]);
+                }
+                else
+                {
+                    extraItems.AddRange(_extraItems);
+                }
+
+                return new CollectionTallyResult(missingItems, extraItems);
             }
         }
 
-        private readonly List<object> _missingItems = new List<object>();
+        private readonly ArrayList _missingItems = new ArrayList();
```

Comment:
Why this change

Reply:
I should have read the description before reviewing

> I do not use IComparable<T> and sort using IList<T>.Sort() because it is difficult to get an IEnumerable into an IList<T>. I took the easy route and use ArrayList.Sort().